docs/sphinx/examples/matrix_multiplication.rst

```diff
@@ -0,0 +1,107 @@
+..
+    Copyright (C) 2021 Dimitra Karatza
+
+    SPDX-License-Identifier: BSL-1.0
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _examples_matrix_multiplication:
+
+===================
+Parallel algorithms
+===================
+
+This program will perform a matrix multiplication in parallel. The output will look something like this:
+
+.. code-block:: text
+
+   Matrix A is : 
+   4 9 6 
+   1 9 8 
+
+   Matrix B is : 
+   4 9 
+   6 1 
+   9 8 
+
+   Resultant Matrix is : 
+   124 93 
+   111 127 
+
+Setup
+=====
+
+The source code for this example can be found here:
+:download:`matrix_multiplication.cpp
+<../../examples/quickstart/matrix_multiplication.cpp>`.
+
+To compile this program, go to your |hpx| build directory (see
+:ref:`hpx_build_system` for information on configuring and building |hpx|) and
+enter:
+
+.. code-block:: shell-session
+
+   $ make examples.quickstart.matrix_multiplication
+
+To run the program type:
+
+.. code-block:: shell-session
+
+   $ ./bin/matrix_multiplication
+
+This should print:
+
+.. code-block:: text
+
+   Matrix A is : 
+   4 9 6 
+   1 9 8 
+
+   Matrix B is : 
+   4 9 
+   6 1 
+   9 8 
+   
+   Resultant Matrix is : 
+   124 93 
+   111 127 
+
+Notice that the numbers may be different because of the random initialization of the matrices.
+
+Walkthrough
+===========
+
+Now that you have compiled and run the code, let's look at how the code works.
+
+First, ``main()`` is used to initialize the runtime system and pass the command line arguments to the program. 
+``hpx::init`` calls ``hpx_main()`` after setting up HPX, which is where our program is implemented.
+
+.. literalinclude:: ../../examples/quickstart/matrix_multiplication.cpp
+   :language: c++
+   :start-after: //[mul_main
+   :end-before: //]
+
+Going on with the ``hpx_main()`` function, we can see that matrix multiplication can be done very easily. 
+
+.. literalinclude:: ../../examples/quickstart/matrix_multiplication.cpp
+   :language: c++
+   :start-after: //[mul_hpx_main
+   :end-before: //]
+
+First, the dimentions of the matrices is defined, so as that the number of columns of matrix A is equal 
+to the number of rows of matrix B. You can freely change the dimentions as you wish. We use standard vectors 
+to define the matrices to be multipled as well as the resultant matrix.
+
+To give some random initial values to our matrices, we use `std::uniform_real_distribution
+<https://en.cppreference.com/w/cpp/numeric/random/uniform_real_distribution>`_. Then, ``hpx::generate()`` is used
+along with ``std::bind()`` as an argument to yield two matrices A and B, which contain values in the range of [0, 10]. 
+You can also freely change the range of the values, if you wish.
```

Comment:
Continuing with the command-line options, at least the seed should also be a command-line option, possibly also the range of values. You can seed the random number generator with something like this: https://github.com/STEllAR-GROUP/hpx/blob/dd3f3d9ae87b93c383877f4f69dc97842c2f8387/libs/core/async_cuda/tests/unit/cublas_matmul.cpp#L269-L273.

Reply:
what is `vm`?

Reply:
`vm` is an object of type `variables_map`. Instead of `hpx_main` taking `argc/argv` you can have it take a `variables_map` instead. To make HPX aware of your custom command line options you have to tell it about them with an `options_description` like here: https://github.com/STEllAR-GROUP/hpx/blob/dd3f3d9ae87b93c383877f4f69dc97842c2f8387/libs/core/async_cuda/tests/unit/cublas_matmul.cpp#L330-L353.